cl/recap/tasks.py

```diff
@@ -474,15 +457,7 @@ def process_recap_zip(self, pk: int) -> dict[str, list[int] | list[Task]]:
         }
 
 
-@app.task(
-    bind=True,
-    autoretry_for=(requests.ConnectionError, requests.ReadTimeout),
-    max_retries=5,
-    interval_start=5 * 60,
-    interval_step=5 * 60,
-    ignore_result=True,
-)
-def process_recap_docket(self, pk):
+async def process_recap_docket(pk):
```

Comment:
> Don't we lose the ability to retry failing tasks (under certain exceptions) with this new approach?

Well, there's other methods of handling exception retries under asyncio. I think doing that at a more granular level is probably better as well.

Reply:
I think I'd rather not go there for now, if we can avoid it. Maybe in a future PR?

Reply:
I mean, the retries are one thing, but taking this out of Celery also means not using the task queue and all it brings to the table?

Reply:
> I mean, the retries are one thing, but taking this out of Celery also means not using the task queue and all it brings to the table?

I think the task queue can be a bit problematic for these upload handlers, for one it seems to make it so that you can't return a proper success/fail response code to the recap extension as you are doing processing(which may fail) after returning a response code to the client. By ensuring the upload and processing is fully complete/successful before returning a success/fail response code we could allow for proper upload retries in the recap extension to then be implemented.

Reply:
I think it might cause reliability issues having a task queue here as it could mask failures.

Reply:
> I think it might cause reliability issues as having a task queue here as it could mask failures.

Reply:
Celery is integrated with Sentry, so we usually get errors that way?

Reply:
Well that's a different issue I think, that's essentially error logging right? I think we want to be able to return success/failure response codes to the client that indicate if the operation fully completed.

Reply:
Hm, it occurs to me that chatting might be faster. Can I invite you to our slack finally?

Reply:
sure

Reply:
sure

Reply:
I think I need your email? Want to put it here and then I'll delete the comment once I've got it?

Reply:
ttys0dev@gmail.com